src/System/refine.rs

```diff
@@ -498,6 +501,10 @@ fn is_new_state<'a>(state_pair: &mut StatePair<'a>, passed_list: &mut Vec<StateP
             panic!("dimensions of dbm didn't match - fatal error")
         }
 
+        if state_pair.has_exceeded_max_bounds() && passed_state_pair.has_exceeded_max_bounds() {
```

Comment:
We should use the built in extrapolation on the dbm instead of doing checks like this. It is sufficient to remove this line and if we apply the extrapolation (rs_dbm_extrapolateMaxBounds) before calling "is_new_state" then it will be caught by is_subset_eq.

has_exceeded_max_bounds also only checks if one clock bound has been exceeded, which isn't enough to consider it a _not a new state_ as they may differ in other clock valuations.